### PR TITLE
chore(serve): webpack-dev-server@3 deprecated warnings

### PR DESCRIPTION
**Pull request by Loonride**
**What kind of change does this PR introduce?**

chore: deprecation warnings

**Did you add tests for your changes?**

Not yet

**If relevant, did you update the documentation?**

No

**Summary**

Adds a warning when user runs `webpack-cli serve` with a webpack-dev-server version below v4. This warning will be pushed to them before and after their webpack compilation to ensure they see it.

Edit: Should I go into greater detail in the warning as to what unexpected behavior may occur?

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: packages/serve/src/index.ts [hunk 1]
```diff
@@ -11,6 +13,10 @@ import argsToCamelCase from './args-to-camel-case';
  * @returns {Function} invokes the devServer API
  */
 export default function serve(...args): void {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const devServerPackage = require('webpack-dev-server/package.json');
```
Comment (evilebottnawi) [inline packages/serve/src/index.ts:17]: try/catch

File: packages/serve/src/index.ts [hunk 2]
```diff
@@ -27,6 +33,23 @@ export default function serve(...args): void {
...
+            const warning = chalk.yellow.bold(
+                'Using webpack-dev-server v3 with ' +
+                    `${chalk.green('webpack-cli serve')} is deprecated and will lead to ` +
+                    'unexpected behavior. Please upgrade to webpack-dev-server v4.',
+            );
+            server.log.warn(warning);
+
+            // send a warning after the compilation to ensure that the message
+            // reaches the user
+            compiler.hooks.done.tap('serve', () => {
```
Comment (evilebottnawi) [inline packages/serve/src/index.ts:48]: Why?

Comment (Loonride) [reply packages/serve/src/index.ts:52]: When the user compiles something, all the compilation stats will be printed and the message will be lost unless it's at the bottom

Comment (Loonride) [reply packages/serve/src/index.ts:52]: dev-server-deprecated-warning

Comment (evilebottnawi) [reply packages/serve/src/index.ts:52]: Interesting place, I think we don't need emit warning(s) to stats in such situation

/cc @webpack/cli-team

Comment (Loonride) [reply packages/serve/src/index.ts:52]: Personally I think it will help the user realize why there might be unexpected behavior.

**Review by snitin315 — CHANGES_REQUESTED** [review]
closing as stale.

**Pull request closed without merging**